backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,85 @@
+import { RedisClient } from 'redis';
+import { CachingObject } from '../data/CachingObject';
+import { Promotion } from '../entity/Promotion';
+import { PromotionDTO } from '../validation/PromotionValidation';
+
+// handles the caching for lat/lon values for restaurants with redis
+export class CachingService {
+  private client: RedisClient;
+
+  // sets redis client for class
+  constructor(client: RedisClient) {
+    this.client = client;
+  }
+
+  // formats promotionDTO for caching
+  async getValuesAndCache(promotion: PromotionDTO): Promise<boolean> {
+    const placeId: string = promotion.placeId;
+    const latitude: number = promotion.lat;
+    const longitude: number = promotion.lon;
+    const result = await this.cacheLatLonValues(placeId, latitude, longitude);
+    return result;
+  }
```

Comment:
I don't think we'll need this method for my suggestion in `PromotionController::addPromotion`